src/compiler/emitter.ts

```diff
@@ -1881,13 +1860,77 @@ var __param = (this && this.__param) || function (paramIndex, decorator) {
                 }
             }
 
+            function emitCallOrNewExpressionTarget(node: Expression): CallTarget {

```

Comment:
I just tested my latest idea. It also requires additional node checks. `emitCallWithSpread()` also emits the accessor and writing the extra node checks is not worth it IMO.  Can't we just rename it to `emitCallTargetAndAccessor()`? It optionally emits and optionally returns the target and accessor. It is the best in term of DRY and also number of operations/comparison for the job.